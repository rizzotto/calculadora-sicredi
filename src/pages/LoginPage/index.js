import React from "react";
import { useDispatch } from "react-redux";
import { useHistory } from "react-router-dom";

import {
    Grid,
    Hidden,
    makeStyles,
    Typography,
    useMediaQuery,
} from "@material-ui/core";
import { Formik, Form, Field } from "formik";
import jwt from "jsonwebtoken";

import { openSnackbar } from "../../actions/snackbarActions.js";
import { ReactComponent as InIcon } from "../../assets/In.svg";
import { ReactComponent as LockIcon } from "../../assets/Lock.svg";
import { ReactComponent as Logo } from "../../assets/LogoSicrediLogin.svg";
import Button from "../../components/Button";
import Container from "../../components/Container";
import Input from "../../components/Input";
import Link from "../../components/Link";
import api from "../../services/api.js";
import theme from "../../styles/customMuiTheme.js";
import homeRoute from "../../utils/homeRoute.js";

const useStyles = makeStyles({
    container: {
        flex: 1,
        height: "100vh",
    },
    text: {
        fontSize: 36,
        color: theme.palette.primary.main,
    },
});

function LoginPage() {
    const classes = useStyles();
    const isMobile = useMediaQuery(theme.breakpoints.down("sm"));
    const history = useHistory();
    const dispatch = useDispatch();

    async function login(values) {
        try {
            const user = await api.post("/login", {
                document: values.cpf,
                password: values.password,
            });
            const token = user.headers.authorization;
            window.localStorage.setItem("token", token.split(" ")[1]);

            const decodedToken = jwt.decode(token.split(" ")[1]);

            history.push(homeRoute(decodedToken.role));
        } catch (e) {
            dispatch(
                openSnackbar({
                    message: "Senha ou CPF incorreto!",
                    severity: "error",
                })
            );
        }
    }

    return (
        <Container maxWidth="lg">
            <Formik
                initialValues={{ cpf: "", password: "" }}
                onSubmit={async (values) => {
                    await login(values);
                }}
            >
                {({ isSubmitting, errors, touched }) => (
                    <Form>
                        <Grid
                            container
                            className={classes.container}
                            justify="center"
                            alignItems="center"
                        >
                            <Hidden smUp>
                                <Grid
                                    item
                                    xs={12}
                                    md={8}
                                    container
                                    justify="center"
                                >
                                    <Logo
                                        width={isMobile ? 312 : 523}
                                        height={isMobile ? 78 : 130}
                                    />
                                </Grid>
                            </Hidden>
                            <Grid item xs={12} md={4} container spacing={4}>
                                <Grid item xs={12}>
                                    <Typography className={classes.text}>
                                        Faça seu login
                                    </Typography>
                                </Grid>
                                <Grid item xs={12}>
                                    <Field
                                        name="cpf"
                                        as={Input}
                                        label="CPF ou Código da Cooperativa"
                                    />
                                </Grid>
                                <Grid item xs={12}>
                                    <Field
                                        name="password"
                                        as={Input}
                                        label="Senha"
                                        type="password"
                                    />
                                </Grid>
                                <Grid item xs={12}>
                                    <Button
                                        fullWidth
                                        type="submit"
                                        disabled={isSubmitting}
                                    >
                                        Entrar
                                    </Button>
                                </Grid>
                                <Grid
                                    item
                                    xs={12}
                                    container
                                    alignItems="center"
                                    spacing={2}
                                >
                                    <Grid item>
                                        <LockIcon width={24} height={24} />
                                    </Grid>
                                    <Grid item>
                                        <Link to="/forgot-password">
                                            Esqueci minha senha
                                        </Link>
                                    </Grid>
                                </Grid>
                                <Grid
                                    item
                                    xs={12}
                                    container
                                    alignItems="center"
                                    spacing={2}
                                >
                                    <Grid item>
                                        <InIcon width={24} height={24} />
                                    </Grid>
                                    <Grid item>
                                        <Link to="/register">
                                            Não tenho cadastro
                                        </Link>
                                    </Grid>
                                </Grid>
                            </Grid>
                            <Hidden smDown>
                                <Grid
                                    item
                                    xs={12}
                                    md={8}
                                    container
                                    justify="center"
                                >
                                    <Logo
                                        width={isMobile ? 312 : 523}
                                        height={isMobile ? 78 : 130}
                                    />
                                </Grid>
                            </Hidden>
                        </Grid>
                    </Form>
                )}
            </Formik>
        </Container>
    );
}

export default LoginPage;
