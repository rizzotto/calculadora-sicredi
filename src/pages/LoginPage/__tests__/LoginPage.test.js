import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

import store from "../../../providers/Store";
import LoginPage from "../index";

jest.mock("../../../assets/Lock.svg", () => {
    return {
        A: true,
        B: () => {
            return <></>;
        },
    };
});

jest.mock("../../../assets/In.svg", () => {
    return {
        A: true,
        B: () => {
            return <></>;
        },
    };
});

jest.mock("../../../assets/LogoSicrediLogin.svg", () => {
    return {
        A: true,
        B: () => {
            return <></>;
        },
    };
});

it("renders LoginPage correctly", () => {
    const tree = renderer
        .create(
            <Provider store={store}>
                <Router>
                    <LoginPage />
                </Router>
            </Provider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
