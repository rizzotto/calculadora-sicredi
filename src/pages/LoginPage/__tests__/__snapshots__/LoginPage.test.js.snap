// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LoginPage correctly 1`] = `
<div
  className="MuiContainer-root makeStyles-container-3 MuiContainer-maxWidthLg"
>
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="MuiGrid-root makeStyles-container-1 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <p
            className="MuiTypography-root makeStyles-text-2 MuiTypography-body1"
          >
            Faça seu login
          </p>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="MuiFormControl-root MuiTextField-root makeStyles-root-4"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink={false}
            >
              CPF ou Código da Cooperativa
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                name="cpf"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-7"
                >
                  <span>
                    CPF ou Código da Cooperativa
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="MuiFormControl-root MuiTextField-root makeStyles-root-4"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink={false}
            >
              Senha
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                name="password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="password"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-7"
                >
                  <span>
                    Senha
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-lightGreen-10 makeStyles-content-11 MuiButton-containedPrimary MuiButton-fullWidth"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              Entrar
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-12"
        >
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <svg
              height={24}
              width={24}
            >
              Lock.svg
            </svg>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <a
              className="makeStyles-link-12"
              href="/forgot-password"
              onClick={[Function]}
            >
              Esqueci minha senha
            </a>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-12"
        >
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <svg
              height={24}
              width={24}
            >
              In.svg
            </svg>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <a
              className="makeStyles-link-12"
              href="/register"
              onClick={[Function]}
            >
              Não tenho cadastro
            </a>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
